model/src/main/java/oracle/kubernetes/weblogic/domain/model/Domain.java

```diff
@@ -460,4 +474,57 @@ public boolean equals(Object other) {
         .append(status, rhs.status)
         .isEquals();
   }
+
+  public List<String> getValidationFailures() {
+    return new Validator().getValidationFailures();
+  }
+
+  class Validator {
+    private List<String> failures = new ArrayList<>();
+    private Set<String> clusterNames = new HashSet<>();
+    private Set<String> serverNames = new HashSet<>();
+
+    List<String> getValidationFailures() {
+      addDuplicateNames();
+      addUnmappedLogHome();
+
+      return failures;
+    }
+
+    private void addDuplicateNames() {
+      getSpec().getManagedServers().stream().map(ManagedServer::getServerName).forEach(this::checkDuplicateServerName);
+      getSpec().getClusters().stream().map(Cluster::getClusterName).forEach(this::checkDuplicateClusterName);
+    }
+
+    private void checkDuplicateServerName(String s) {
+      if (serverNames.contains(s))
+        failures.add(String.format("More than one server is named '%s'", s));
+      else
+        serverNames.add(s);
+    }
+
+    private void checkDuplicateClusterName(String s) {
+      if (clusterNames.contains(s))
+        failures.add(String.format("More than one cluster is named '%s'", s));
+      else
+        clusterNames.add(s);
+    }
+
+    private void addUnmappedLogHome() {
+      if (!isLogHomeEnabled()) return;
+
+      if (getSpec().getAdditionalVolumeMounts().stream().map(V1VolumeMount::getMountPath).noneMatch(this::mapsLogHome))
+        failures.add("No volume mount contains path for log home");
```

Comment:
for log home --> 'for the logHome '%s' specified in the domain resource' (%s should be this::mapsLogHome).

Reply:
Is this a valid check?  IIRC, the startup code does a 'mkdir -p'?  If it does, then maybe only check that the first directory in each path matches.

Reply:
this::mapsLogHome is a boolean. Do you mean this::logHome ?

Reply:
Yes.

Reply:
Doing a `mkdir -p` shouldn't matter in this case, as we are not looking at the actual file system, here. The key is that:

`/my/path/to/log` is not a parent of `/my/path/to/logfiles/mydomain` even though the "startsWith" check without adding the trailing separator would pass. That's why we have to ensure that the trailing separator is there.

Reply:
Ah.  Makes sense.  Please update the log message regardless.